bundles/org.openhab.binding.paradoxalarm/src/main/java/org/openhab/binding/paradoxalarm/internal/communication/Main.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.paradoxalarm.internal.communication;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.openhab.binding.paradoxalarm.internal.exceptions.ParadoxBindingException;
+import org.openhab.binding.paradoxalarm.internal.model.ParadoxPanel;
+import org.openhab.binding.paradoxalarm.internal.model.Partition;
+import org.openhab.binding.paradoxalarm.internal.model.RawStructuredDataCache;
+import org.openhab.binding.paradoxalarm.internal.model.Zone;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link Main} - used for testing purposes of low-level stuff.

```

Comment:
Should it be in a testing package then?

Reply:
Does it belong in this PR? This does not follow the guidelines of the code so puts out a very wrong example for other binding developers. We should not catch generic exceptions, provide context when logging, never use system exit. Please let me know what the target audience is for this class

Reply:
The target for this class is to be used for testing the low-level live communication. That's why all the ugly mechanics. I've had the same proposal by Hilbrand but I would strongly prefer to keep this class just for testing outside of openhab environment. Of course probably I can try to keep it in separate branch but that would be overhead. Unless it's strictly forbidden to use main class for testing I would like to keep it. Please let me know if we can agree on keeping it.

Reply:
We the new build system it's now allowed to have a test folder `/src/test/java` in the same project. Maybe you can move this file to that location so it's outside the main code.

Reply:
Now when looking into the options for fixing Thread.sleep() requirement by Martin, I believe I will have to get rid of main method simply because I will need to make dependency to ThreadPoolManager in my own model... I don't like this at all because I wanted to keep the low-level part simple and loosly coupled from the OpenHAB framework. Maybe I can check if scheduler is null to create my own implementation but I'm not sure if it worths the effort just for the sake of loose coupling of the model. What do you think @Hilbrand  @martinvw ?

Reply:
Maybe I lost a little bit of recent updates but afaik all you need is the scheduler of the parent class of the handler. If really need you could get your instance of ScheduledExecutorService from the jdk for your main class.

@Hilbrand please correct me if I’m wrong / outdated.

Reply:
Yeah it's done in BaseThingHandler. There is initialized the scheduler using static call to this: ThreadPoolManager
            .getScheduledPool(THING_HANDLER_THREADPOOL_NAME). Probably somewhere I need to provide the executor service to my communicator and in main class file I'll see how to get it from jdk. Will look into that further tomorrow. Thanks a lot for the feedback ! Most of feedback is valuable for me. It's good that I'm home these days so I can have a look into the feedback and do the fixes faster than normal...